figures/sync_architecture.xml

```diff
@@ -0,0 +1 @@
+<mxfile type="google" userAgent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36" version="6.3.4" editor="www.draw.io"><diagram name="Page-1">7Vtbc9o4FP41zOw+wEiWbcxjSNP2oTuTabqz3aeMYgvjjbGoLELor1/JSL7JppAIApMmD5hjWZLP5fvOkcQAXS+ePzG8nP9FI5IOHBA9D9CHgeNAfxKIDynZKAn00VYSsyRSskpwl/wkSgiUdJVEJG805JSmPFk2hSHNMhLyhgwzRtfNZjOaNkdd4pgYgrsQp6b0nyTi863UC9xK/pkk8VyN7AZq3g84fIwZXWVquIGDZsXf9vYC665U+3yOI7quidDNAF0zSvn2avF8TVKpW601/Rzf6KkO0HTOF6n4AsVlcftjz8Nwn4fFyzGS8fpwvf2hsdEjiYQi1VfK+JzGNMPpTSWdFuohsgvQHP4/wvlG+QJecSpEVQ9fKF2qdjln9JFc05SyYkwUAPnf+QL6lemKhWqG/mQrkzOtNVKv+InQBeFsIxowkmKePDVdAivPist25aO3NBHDOkCFga+dQgWB6wfNLjhmMeHqqUrN4qI2jUpUKL/bEGPTLU5gB/Kc8O/yegQgUt//ld2NPPXtlrBEvAZhapDXWg6cxnIlDB1ouSvG8KbWbCkb5DvGAd3jVI6w7fGlbqH19YTTldLFbJ5vsvD+x4oISdtnngjjiQDBL/iBpLc0T3hCM3HrgXJOF00n0W2v0iSWbbh0i5aBM5pJR5slaVqzOQDTm48SeQT4LeXIi+dY0sgI/1wxMiqmdh+TTPhOuMsz5AzI804/UHeH7qSpZ1fZc11hO9TKmtdxHfS7TiNcd8UmshmbQglsI4NuCEbA1YJa1Mmv7bA7FbA6iqmL6FBv79gO2X31rmdjOr/g6HceAMjzThcAsCNpuYAsAb0N15QRYz9LgB0J4G9DlDGCxi1LOGAvSxzK+gh2DmOL9KFjAt8dwSycG8YX6MGbNmUkT37ih6KBtJB6F9Hamw68D0KCFdyFwmaSYAwcXCRRVDhOKkF0WhZENQdQJVGnA/SCW1nsqekN6hVTJ+iBEfTdhq6HSjWvdRUHOE0bNnugs1lOXhut2u9N/lqSLEqy+J1zmOM5p+Qwq5VumcWBZgY3DoK3zuEmHTmc91Y5HHQCE8tECAwcP5XIFSVP4jKWl19JSFmU6zui59pNq8iXkhl/Ge59LP56cG9sEfeQA5rBAq3gHnS7OrWKetD1DIv/vYwwN4HubWy4k7us2VCYMGhRlx0TDiE6PnNNeoIW3DIakjwXujxL1lpT9kjYPaOFD1jgLAc2c72uhQfvaJTl2aSsE/HPGNrmmv0WZ/ctuw5N9n0Xdo5jL9v3jWC7ZqTAS4Mjb7eJo2h9CWy5E2k1S9hgSw+NgyOw5TBo9jo8Bl/qeKmZv7Lye4fbrhLhaHCLrMJtX4XgQueNKwQNlGdSISAT/67Cx4yu02Iyl1ko7II+ZA36wEjYvhkxtpJM0AmodpHPXOYShv+Nep2L+8dCvbFZralU/wPmWHzc6TjLlzjTUcYpI7X4q986T5uJ+hM/4NySwVyvmXsibYmawXxoGizwXm+wYHIUlpK7kRUzncd2pC6g60R1pBMEh1YFwxK9unf+D37g9XUEMFfarqJIRvA2nr/h/DE/N3psRfHNlT/1/QM9R0eElVIClOtXtpdtPKezX6uUqiPGRPO7cE6iVUreC6cOXdSKt66Vmw6MtlJKgEvZpz36nqzXXqU5zpbssCSGFiX347HfPbPeB3y0q30Tv3+9N43afrYlPiunBFz3QryvTvYT5208Eo39M/VIPTNrGYLbUWkXGcJZpgYdXqMd286iodc6x2HrbMG4u1urRK+jxST6z9++3coPnEXviu1bJbRjVmTH26exekD0ktneOIEF9qN7c9cF/aKjHro8FKVhz4T75vWL9hZ2g8wM/ivBEqTLZU+LMH20M2D66JOVM2CtiswSTHutI2BHwWmIdi/XKPi0DwYlAkc4nxc9d/8+qCMTgzA4DVz4gd+wgLfnec2X2AH2Hmko4+o3dW6b69ziFNSpN9drZjH0vp4nnNwtceGfa6GLpqoN1991SOtQ7fhOy0WhmVeUx+Pr2vFtaAeaS3rN1fiamvwfK/mbw2n1A8ZhuNXJlYzU+OEPR24riGmJcUHr+s/qeb2Of73KhVfLMAAS/kkWXdSCf5jSVWTJB9wm58OxGSCTjpWkF7iA+Fr9dHQLc9Xvc9HN/w==</diagram></mxfile>
```

Comment:
@aidenkeating 1 comment on above diagram. I believe the sync records http handler doesn't read directly from the Dataset Backend. It reads from an in-MongoDB cache of the records from the last sync loop.